index.js

```diff
@@ -0,0 +1,84 @@
+const Base = require('bfx-facs-base')
+const { HyperCoreLogger } = require('hypercore-logs')
+const HyperCoreTransport = require('./lib/winston-transport-hypercore')
+const path = require('path')
+const winston = require('winston')
+
+class HyperCoreLoggerFacility extends Base {
+  constructor (caller, opts, ctx) {
+    super(caller, opts, ctx)
+
+    this.name = 'hypercore-logger'
+    this._hasConf = true
+
+    this.init()
+  }
+
+  _start0 (cb) {
+    super._start0(err => {
+      if (err) {
+        return cb(err)
+      }
+
+      const hypercoreCfg = this.conf
+      if (!hypercoreCfg?.enable) {
+        return cb(null)
+      }
+
+      const feedDir = path.join(process.cwd(), hypercoreCfg.storageDir)
+      const server = new HyperCoreLogger(
+        feedDir,
+        hypercoreCfg.publicKey,
+        {
+          secretKey: hypercoreCfg.secretKey
+        }
+      )
+
+      this.hyperCoreLogs = server
+
+      server.start()
+        .then(() => {
+          if (!hypercoreCfg.publicKey) {
+            console.log(`HyperCore public key: ${server.feed.key.toString('hex')}`)
+          }
+
+          cb(null)
+        })
+        .catch(err => cb(err))
+    })
+  }
+
+  getLogger () {
+    const { format } = winston
```

Comment:
tbh I'm a bit against using winston even though it's one of most known packages, we can simply build something like this:
```js
const { format } = require('util')
...
log (...params) {
   const line = format(new Date().toIsoString(), ...params)
   this.server.feed.append(line)
   console.log(line)
}

debug (...params) {
  if (this.ctx.caller.debug < 1) return

   const line = format(new Date().toIsoString(), ...params)
   this.server.feed.append(line)
   console.log(line)
}

error (...params) {
   const line = format(new Date().toIsoString(), ...params)
   this.server.feed.append(line)
   console.error(line)
}
```